custom_train.py

```diff
@@ -0,0 +1,168 @@
+        self._TPL_CYCLES = config.getint('TRIPLET_TRAINING', 'CYCLES')
+        self._TPL_EPOCHS = config.getint('TRIPLET_TRAINING', 'EPOCHS')
+        self._TPL_FILTER_SIZE = config.getint('TRIPLET_TRAINING', 'FILTER_SIZE')
+        self._TPL_MARGIN = config.getfloat('TRIPLET_TRAINING', 'MARGIN')
+        self._TPL_OPTIMIZER = config.get('TRIPLET_TRAINING', 'OPTIMIZER')
+        self._TPL_LR_VALUE = config.getfloat('TRIPLET_TRAINING', 'LR_VALUE')
+        self._TPL_FREEZE_VARS = [
+            var.replace(" ", "") for var in
+            config.get('TRIPLET_TRAINING', 'FREEZE_VARS').split(',')
+        ]
+
+        # Throw exception if optimizer is not defined
+        if self._CLS_OPTIMIZER not in self._OPTIMIZER_MAP.keys():
+            raise ValueError("CLASSIFIER_TRAINING OPTIMIZER not defined.")
+        if self._TPL_OPTIMIZER not in self._OPTIMIZER_MAP.keys():
+            raise ValueError("TRIPLET_TRAINING OPTIMIZER not defined.")
+
+    def train_classifier(self):
+        '''Train classifer according to specs in config file.'''
+        self.datset_builder.ONE_HOT = True
+        model, input_name = self.model_builder.create_full_model()
+        model.summary()
+
+        boundaries = self._CLS_LR_BOUNDARIES
+        values = self._CLS_LR_VALUES
+        lr_schedule = self._LR_SCHEDULE_MAP['PiecewiseConstantDecay'](
+            boundaries=boundaries, values=values)
+        optimizer = self._OPTIMIZER_MAP[self._CLS_OPTIMIZER](
+            learning_rate=lr_schedule)
+
+        loss = self._LOSS_MAP['CrossEntropy'](from_logits=True)
+
+        accuracy = self._METRIC_MAP['Accuracy']()
+        model.compile(optimizer=optimizer, loss=loss, metrics=[accuracy])
+
+        estimator = tf.keras.estimator \
+            .model_to_estimator(keras_model=model, model_dir=self._CLS_CKPT_DIR)
+        train_spec = tf.estimator.TrainSpec(
+            input_fn=self.datset_builder.get_train_input_fn(input_name),
+            max_steps=self._CLS_MAX_STEP)
+        eval_spec = tf.estimator.EvalSpec(
+            input_fn=self.datset_builder.get_eval_input_fn(input_name))
+
+        tf.estimator.train_and_evaluate(estimator, train_spec, eval_spec)
+
+    def train_triplet_transfer(self):
+        """Train encoder with triplet loss according to specs in config file."""
+        self.datset_builder.ONE_HOT = False
+        model, input_name = self.model_builder.create_full_model()
+        model.summary()
+
+        optimizer = self._OPTIMIZER_MAP[self._TPL_OPTIMIZER](self._TPL_LR_VALUE)
+
+        # Load initial weights from self._TPL_INIT_DIR
+        init_dir = self._TPL_INIT_DIR
+        latest = tf.train.latest_checkpoint(init_dir)
+        model.load_weights(latest)
+
+        freeze_var_names = self._TPL_FREEZE_VARS
+        freeze_layers = [layer for layer in model.layers[0].layers if
+                         layer.name[:5] in freeze_var_names]
+        for layer in freeze_layers:
```

Comment:
Add comments about what you are doing here. I remember you explain to me why you freeze them here, but write them into comments.